core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -3164,7 +3166,7 @@ private Type coerceToSingleType(StackableAstVisitorContext<Context> context, Nod
                 return superType;
             }
 
-            throw semanticException(TYPE_MISMATCH, node, message, firstType, secondType);
+            throw semanticException(TYPE_MISMATCH, node, "%s", format(message, firstType, secondType));
```

Comment:
is it because one `@FormatMethod` cannot delegate to other `@FormatMethod`?
if so, seems like error-prone's bug to me

Reply:
It can, but looks like it gets confused by them being different types. I wasn't sure if this is a bug or a weird corner case.

Reply:
it doesn't look like we were doing anything wrong here

Reply:
The actual message is:

```
java: [FormatStringAnnotation] The format argument types passed with an @FormatString must match the types of the format arguments in the @FormatMethod header where the format string was declared.
  	Format arg types passed: [io.trino.spi.type.Type, io.trino.spi.type.Type]
  	Format arg types expected: [io.trino.sql.tree.Expression, io.trino.sql.tree.Expression]
```

Thing is, the message is the same but the message arguments passed to `semanticException` are not the same as the parameters of the enclosing method. So it kinda sorta makes sense?